src/py21cmfast_tools/lc2ps.py

```diff
@@ -264,23 +351,55 @@ def log_bin(ps, kperp, kpar, bins=None):
         The number of bins or the bin edges to use for binning the kpar axis.
         If None, produces 16 bins logarithmically spaced between
         the minimum and maximum `kpar` supplied.
+    interp : str, optional
+        If 'linear', use linear interpolation to calculate the PS at the specified
+        kpar bins.
 
     """
     if bins is None:
-        bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), 17)
+        if kpar_log:
+            bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), 17)
+        else:
+            bins = np.linspace(kpar[0], kpar[-1], 17)
     elif isinstance(bins, int):
-        bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), bins + 1)
+        if kpar_log:
+            bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), bins + 1)
+        else:
+            bins = np.linspace(kpar[0], kpar[-1], bins + 1)
     elif isinstance(bins, (np.ndarray, list)):
         bins = np.array(bins)
     else:
         raise ValueError("Bins should be np.ndarray or int")
     modes = np.zeros(len(bins) - 1)
     new_ps = np.zeros((len(kperp), len(bins) - 1))
-    for i in range(len(bins) - 1):
-        m = np.logical_and(kpar >= bins[i], kpar < bins[i + 1])
-        new_ps[:, i] = np.nanmean(ps[:, m], axis=1)
-        modes[i] = np.sum(m)
-    bin_centers = np.exp((np.log(bins[1:]) + np.log(bins[:-1])) / 2)
+    if interp == "linear":
+        if kpar_log:
+            bin_centers = np.exp((np.log(bins[1:]) + np.log(bins[:-1])) / 2)
+        else:
+            bin_centers = (bins[1:] + bins[:-1]) / 2
```

Comment:
this can just be moved outside the if-statement and the following redundant code removed

Reply:
Good point, I agree.